api-types.ts

```diff
@@ -9,25 +9,77 @@ export type Scalars = {
   Boolean: boolean;
   Int: number;
   Float: number;
+  Date: any;
 };
 
+
 export type Group = {
   __typename?: 'Group';
   id?: Maybe<Scalars['Int']>;
   name?: Maybe<Scalars['String']>;
 };
 
+export type GroupInput = {
+  name?: Maybe<Scalars['String']>;
+};
+
 export type Query = {
   __typename?: 'Query';
   listGroups: Array<Maybe<Group>>;
+  listShipments: Array<Maybe<Shipment>>;
 };
 
 export type Mutation = {
   __typename?: 'Mutation';
-  addGroup?: Maybe<Group>;
+  addGroup: Group;
+  addShipment: Shipment;
 };
 
 
 export type MutationAddGroupArgs = {
-  name?: Maybe<Scalars['String']>;
+  input: GroupInput;
+};
+
+
+export type MutationAddShipmentArgs = {
+  input: ShipmentInput;
+};
+
+export type ShipmentInput = {
+  shippingRoute: ShippingRoute;
+  labelYear: Scalars['Int'];
+  labelMonth: Scalars['Int'];
+  sendingHubId: Scalars['Int'];
+  receivingHubId: Scalars['Int'];
+  status: ShipmentStatus;
+};
+
+export enum ShippingRoute {
+  Uk = 'UK'
+}
+
+export enum ShipmentStatus {
+  Announced = 'ANNOUNCED',
+  Open = 'OPEN',
+  Staging = 'STAGING',
+  InProgress = 'IN_PROGRESS',
+  Complete = 'COMPLETE',
+  Abandoned = 'ABANDONED'
+}
+
+export type Shipment = {
+  __typename?: 'Shipment';
+  id?: Maybe<Scalars['Int']>;
```

Comment:
What is the advantage over using:

```typescript
     id?: number
```

Reply:
In addition, this semantically means, that there maybe maybe an int, because the the property is marked optional using the TypeScript `?` and even if the value is present, it could not contain an id. Why these duplicate semantics?

Reply:
Yeah, this file is generated by [`graphql-codegen`](https://graphql-code-generator.com/), which is where these `Maybe` types come from. Getting our `schema.graphql` in the shape we want is what will influence the output here. I will mark `id` as non-null with `id: Int!` which should shore some of this up.

- [ ] TODO: Make `id` `Int!` in `schema.graphql`